### DISCUSSION TITLE
Add package as dependency for another package

### DISCUSSION THREAD

**Discussion by mkamalza** [applejag/Newtonsoft.Json-for-Unity#92] [Q&A]
Hi,

Is it possible to add this package as a dependency for another package through adding it to the latter's **package.json** file?  I have tried a few ways, including the scoped registry method but it does not work.

**Comment by applejag** [comment]
Hello @mkamalza!

As the Newtonsoft.Json scene looks like it does in the Unity community right now, with so many scattered versions, I would strongly advise against locking down to 1 specific version.

For reference, I have identified these 5 different sources so far: (there are probably more)

- Old old deprecated NuGet package: <https://www.nuget.org/packages/Unity.Newtonsoft.Json/>
- The most used Json.NET version of all (the one from Assets Store): <https://www.parentelement.com/assets/json_net_unity>
- Predecessor of this repo (I got a lot of inspiration from this one): <https://github.com/SaladLab/Json.Net.Unity3D#readme>
- This repo: <https://github.com/jilleJr/Newtonsoft.Json-for-Unity#readme>
- Unity's own fork of this repo, hosted in their official package repo: <https://docs.unity3d.com/Packages/com.unity.nuget.newtonsoft-json@2.0/manual/index.html>

The best recommendation I have is to add an installation guide to your package that says something like:

> "Prerequisite: Add Newtonsoft.Json. Recommendation: https://github.com/jilleJr/Newtonsoft.Json-for-Unity"

It's not a pretty solution, but as there is no de-facto package out there, and the fact that Unity does not want to embrace for example this repo but instead try to introduce their own package in the already extensive list of Json.NET alternatives, then this issue remains.

Doing a search on GitHub after this package, it's clear that a lot of users don't even use UPM but instead embed the binaries directly into their repo.

To answer your question directly though, no I've actually never tried to have any dependencies in a package on other packages from other scoped registries. Reading the docs (https://docs.unity3d.com/Manual/upm-manifestPkg.html) I interpret it such that if you add a dependency on my package `jillejr.newtonsoft.json-for-unity`, then the user that uses your package must add a scoped registry in their project `manifest.json` so their Unity instance can resolve the indirect dependency on `jillejr.newtonsoft.json-for-unity`
